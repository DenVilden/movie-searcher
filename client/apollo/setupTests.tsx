import { render } from '@testing-library/react';
import { MockedProvider, MockedResponse } from '@apollo/client/testing';
import { ThemeProvider } from 'styled-components';
import { theme } from '../pages/_app';

// eslint-disable-next-line jest/prefer-spy-on
HTMLElement.prototype.scrollIntoView = jest.fn();

type RenderApolloOptions = {
  mocks?: MockedResponse[];
  addTypeName?: boolean;
};

export const renderApollo = (
  element: React.ReactElement,
  { mocks = [], addTypeName = false }: RenderApolloOptions = {},
) =>
  render(
    <MockedProvider addTypename={addTypeName} mocks={mocks}>
      <ThemeProvider theme={theme}>{element}</ThemeProvider>
    </MockedProvider>,
  );

export * from '@testing-library/react';
