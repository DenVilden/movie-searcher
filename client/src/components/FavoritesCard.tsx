import {
  CardActionArea,
  LinearProgress,
  Typography,
  CardMedia,
} from "@material-ui/core";
import { useRouter } from "next/router";
import styled from "styled-components";
import ErrorMessage from "../containers/ErrorMessage";
import { useGetMovieInfoQuery } from "../graphql";

const StyledTypography = styled(Typography)`
  padding: ${(props) => props.theme.spacing(2)}px;
`;

const CardWrapper = styled.div`
  display: flex;
  min-width: 200px;
`;

const StyledCardMedia = styled(CardMedia)`
  display: none;
  height: 56px;
  margin: ${(props) => props.theme.spacing(1)}px;
  width: 50px;

  ${(props) => props.theme.breakpoints.up("sm")} {
    display: block;
  }
`;

type Props = {
  toggleFavoritesOpen: () => void;
  id: string;
};

const FavoritesCard = ({ toggleFavoritesOpen, id }: Props) => {
  const { loading, error, data } = useGetMovieInfoQuery({ variables: { id } });

  const router = useRouter();

  if (error) return <ErrorMessage error={error} />;

  if (loading || !data?.movieInfo) return <LinearProgress />;

  return (
    <CardActionArea
      data-testid="favorites-card"
      onClick={() => {
        router.push(`/movie/${id}`);
        toggleFavoritesOpen();
      }}
    >
      <CardWrapper>
        <StyledCardMedia
          image={data.movieInfo.poster_path || "/no-image.jpg"}
          src="img"
          title={data.movieInfo.title}
        />
        <StyledTypography>{data.movieInfo.title}</StyledTypography>
      </CardWrapper>
    </CardActionArea>
  );
};

export default FavoritesCard;
