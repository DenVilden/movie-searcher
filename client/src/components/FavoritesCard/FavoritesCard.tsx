import React from 'react';
import { CardActionArea } from '@material-ui/core';
import { useHistory } from 'react-router-dom';
import noImage from '../../assets/no-image.jpg';
import {
  StyledTypography,
  CardWrapper,
  StyledCardMedia,
} from './FavoritesCard.styles';
import { MovieInfo } from '../../__generated__/types';

type Props = {
  toggleFavoritesOpen: () => void;
  movie: MovieInfo;
};

const FavoritesCard = ({ toggleFavoritesOpen, movie }: Props) => {
  const history = useHistory();

  return (
    <CardActionArea
      data-testid="favorites-card"
      onClick={() => {
        history.push(`/movie/${movie.id}`);
        toggleFavoritesOpen();
      }}
    >
      <CardWrapper>
        <StyledCardMedia
          image={movie.poster_path || noImage}
          src="img"
          title={movie.title}
        />
        <StyledTypography>{movie.title}</StyledTypography>
      </CardWrapper>
    </CardActionArea>
  );
};

export default FavoritesCard;
