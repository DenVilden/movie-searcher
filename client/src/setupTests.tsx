import React from 'react';
import { render } from '@testing-library/react';
import { MockedProvider, MockedResponse } from '@apollo/client/testing';
import { ThemeProvider } from 'styled-components';
import { theme } from './pages/_app';
import defaultResolvers from './graphql/resolvers';

window.HTMLElement.prototype.scrollIntoView = jest.fn();

type RenderApolloOptions = {
  mocks?: MockedResponse[];
  addTypeName?: boolean;
  cache?: any;
  resolvers?: any;
};

export const renderApollo = (
  element: React.ReactElement,
  { mocks, addTypeName, resolvers, cache }: RenderApolloOptions = {}
) =>
  render(
    <MockedProvider
      addTypename={addTypeName}
      cache={cache}
      mocks={mocks}
      resolvers={resolvers || defaultResolvers}
    >
      <ThemeProvider theme={theme}>{element}</ThemeProvider>
    </MockedProvider>
  );

export * from '@testing-library/react';
