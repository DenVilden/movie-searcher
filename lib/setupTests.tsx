/* eslint-disable import/no-extraneous-dependencies */
import { render } from '@testing-library/preact'
import { MockedProvider, MockedResponse } from '@apollo/client/testing'
import { ThemeProvider, createMuiTheme } from '@material-ui/core/styles'
import '@testing-library/jest-dom/extend-expect'

const theme = createMuiTheme()

type RenderApolloOptions = {
  addTypeName?: boolean
  mocks?: MockedResponse[]
}

export function renderApollo(
  element: React.ReactElement,
  { mocks = [], addTypeName = false }: RenderApolloOptions = {},
) {
  return render(
    <MockedProvider addTypename={addTypeName} mocks={mocks}>
      <ThemeProvider theme={theme}>{element}</ThemeProvider>
    </MockedProvider>,
  )
}

export { screen, fireEvent } from '@testing-library/preact'
