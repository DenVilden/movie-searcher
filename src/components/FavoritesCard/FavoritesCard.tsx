import React from 'react';
import { CardActionArea } from '@material-ui/core';
import noImage from '../../assets/no-image.jpg';
import {
  StyledTypography,
  CardWrapper,
  StyledCardMedia,
} from './FavoritesCard.styles';

type Props = {
  clearInputValue: () => void;
  toggleFavoritesOpen: () => void;
  poster: string | null;
  title: string;
  goTo: () => void;
};

const FavoritesCard = ({
  clearInputValue,
  toggleFavoritesOpen,
  poster,
  title,
  goTo,
}: Props) => (
  <CardActionArea
    onClick={() => {
      goTo();
      clearInputValue();
      toggleFavoritesOpen();
    }}
  >
    <CardWrapper>
      <StyledCardMedia image={poster || noImage} src="img" title={title} />
      <StyledTypography>{title}</StyledTypography>
    </CardWrapper>
  </CardActionArea>
);

export default FavoritesCard;
