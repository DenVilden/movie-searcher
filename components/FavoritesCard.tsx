import styled from '@emotion/styled'
import { MenuItem, Typography } from '@material-ui/core'
import Image from 'next/image'
import Link from 'next/link'

import { Favorite } from '@/apollo/client'

const Title = styled(Typography)`
  padding-left: 15px;
  white-space: normal;
  width: 100%;
`

interface Props {
  favorite: Favorite
  handleToggle: () => void
}

export default function FavoritesCard({
  favorite: { id, media_type, poster_path, title },
  handleToggle,
}: Props) {
  return (
    <Link href={`/${media_type}/${id}`}>
      <MenuItem onClick={handleToggle}>
        <Image
          alt={title}
          height={50}
          layout="fixed"
          objectFit="cover"
          src={poster_path || '/no-image.png'}
          width={50}
        />
        <Title>{title}</Title>
      </MenuItem>
    </Link>
  )
}
