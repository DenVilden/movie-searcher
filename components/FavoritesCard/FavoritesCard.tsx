import { CardActionArea, Typography } from '@material-ui/core';
import { useRouter } from 'next/router';
import Image from 'next/image';
import styled from '@emotion/styled';
import { Favorite } from '../../apollo';

const StyledTypography = styled(Typography)`
  padding: ${(props) => props.theme.spacing(3, 2, 2, 2)};
`;

const CardWrapper = styled.div`
  display: flex;
  min-width: 200px;
`;

const ImageWrapper = styled.div`
  display: none;
  height: 56px;
  margin: ${(props) => props.theme.spacing(1)};
  position: relative;
  width: 50px;

  ${(props) => props.theme.breakpoints.up('sm')} {
    display: block;
  }
`;

interface Props {
  handleToggle: () => void;
  favorite: Favorite;
}

export default function FavoritesCard({
  handleToggle,
  favorite: { media_type, id, title, poster_path },
}: Props) {
  const router = useRouter();

  return (
    <CardActionArea
      onClick={() => {
        router.push(`/${media_type}/${id}`);
        handleToggle();
      }}
    >
      <CardWrapper>
        <ImageWrapper>
          <Image
            layout="fill"
            alt={title}
            src={poster_path || '/no-image.jpg'}
          />
        </ImageWrapper>
        <StyledTypography>{title}</StyledTypography>
      </CardWrapper>
    </CardActionArea>
  );
}
