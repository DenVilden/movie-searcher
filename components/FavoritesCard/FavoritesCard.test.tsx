import { renderApollo } from '../../lib/setupTests';
import FavoritesCard from './FavoritesCard';

const mock = {
  id: 1,
  backdrop_path: null,
  poster_path: null,
  title: 'test',
  overview: 'test data',
  budget: '0',
  revenue: '0',
  vote_average: 5,
  release_date: '2020',
  similar: {
    results: [
      {
        id: 1,
        title: 'test',
        release_date: '2020',
        poster_path: null,
      },
    ],
  },
};

describe('favoritesCard', () => {
  it('should take a snapshot', () => {
    const { asFragment } = renderApollo(
      <FavoritesCard handleToggle={jest.fn()} favorite={mock} />,
    );

    const element = asFragment();

    expect(element).toMatchSnapshot();
  });
});
