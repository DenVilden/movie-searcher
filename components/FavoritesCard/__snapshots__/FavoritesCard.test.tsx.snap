// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`favoritesCard should take a snapshot 1`] = `
<DocumentFragment>
  <button
    class="MuiButtonBase-root MuiCardActionArea-root"
    data-testid="favorites-card"
    tabindex="0"
    type="button"
  >
    <div
      class="FavoritesCard__CardWrapper-sc-13xt0o0-1 eUwyVa"
    >
      <div
        class="FavoritesCard__ImageWrapper-sc-13xt0o0-2 uhnzZ"
      >
        <div
          style="display: block; overflow: hidden; position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; margin: 0px;"
        >
          <img
            alt="test"
            decoding="async"
            src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
            style="visibility: hidden; position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
          />
        </div>
      </div>
      <p
        class="MuiTypography-root FavoritesCard__StyledTypography-sc-13xt0o0-0 ktwyNo MuiTypography-body1"
      >
        test
      </p>
    </div>
    <span
      class="MuiCardActionArea-focusHighlight"
    />
    <span
      class="MuiTouchRipple-root"
    />
  </button>
</DocumentFragment>
`;
